test/SlowTests/Server/Documents/DataArchival/DataArchivalIndexingTests.cs

```diff
@@ -0,0 +1,483 @@
+                new IndexDefinition
+                {
+                    Maps = {"from o in docs where o[\"@metadata\"][\"@archived\"] == true select new" +
+                        "{" +
+                        "    Name = o.Name" +
+                        "}"},
+                    Name = "test",
+                    SourceItemKind = IndexSourceItemKind.WithArchived
+                }}));
+
+
+            // Insert document with archive time before activating the archival
+            var company = new Company {Name = "Company Name", Address1 = "Dabrowskiego 6"};
+            var retires = SystemTime.UtcNow.AddMinutes(5);
+            using (var session = store.OpenAsyncSession())
+            {
+                await session.StoreAsync(company);
+                var metadata = session.Advanced.GetMetadataFor(company);
+                metadata[Constants.Documents.Metadata.ArchiveAt] = retires.ToString(DefaultFormat.DateTimeOffsetFormatsToWrite);
+                await session.SaveChangesAsync();
+            }
+
+            // Make sure that the company is skipped from indexing by map
+            using (var session = store.OpenAsyncSession())
+            {
+                await Indexes.WaitForIndexingAsync(store);
+                var entries = await session.Query<User>("test").Customize(x => x.WaitForNonStaleResults()).ToListAsync();
+                Assert.Equal(0, entries.Count);
+            }
+            
+            // Activate the archival
+            await SetupDataArchival(store);
+            var database = await Databases.GetDocumentDatabaseInstanceFor(store);
+            database.Time.UtcDateTime = () => DateTime.UtcNow.AddMinutes(10);
+            var documentsArchiver = database.DataArchivist;
+            await documentsArchiver.ArchiveDocs();
+            
+            // Make sure that the company is not skipped from indexing by map
+            using (var session = store.OpenAsyncSession())
+            {
+                await Indexes.WaitForIndexingAsync(store);
+                var entries = await session.Query<User>("test").Customize(x => x.WaitForNonStaleResults()).ToListAsync();
+                Assert.Equal(1, entries.Count);
+            }
+        }
+    }
+    
+    
+    private class CountersIndex : AbstractCountersIndexCreationTask<Employee>
+    {
+        public class Result
+        {
+            public string CompanyName { get; set; }
+        }
+
+        public CountersIndex()
+        {
+            AddMap("Companies",
+                counters => from counter in counters
+                            select new Result
+                            {
+                                CompanyName = LoadDocument<Company>(counter.Name).Name
+                            });
+        }
+    }
+    
+    [RavenTheory(RavenTestCategory.Indexes)]
+    [RavenData(SearchEngineMode = RavenSearchEngineMode.All)]
+    public async Task CanArchiveEntity_WillNotBeRelevantInCountersIndex(Options options)
```

Comment:
All test names here starts with `CanArchiveEntity_` - what was the intention? I don't find it to be very meaningful